src/hooks/UseDynamicHighilghtedLink.js

```diff
@@ -0,0 +1,140 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
```

Comment:
Should the name of the file have a capital letter?

Reply:
You're right, it should actually start by a lower case letter, I will change that.